yare-core/src/main/java/com/sabre/oss/yare/core/reference/ValueConverter.java

```diff
@@ -30,11 +30,14 @@
 import com.sabre.oss.yare.core.model.Rule;
 
 import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
 import java.util.Optional;
 
 public class ValueConverter<R> {
     private static final ChainedTypeExtractor chainedTypeExtractor = new ChainedTypeExtractor();
-    private static final String CONTEXT_PATH = "ctx";
+    private static final List<String> SPECIAL_REFERENCES = Collections.unmodifiableList(Arrays.asList("ctx", "engineController"));
```

Comment:
What about `RESERVED_REFERENCES` convention instead of `SPECIAL_REFERENCES`? I think it is more meaningful

Reply:
Done.